hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3065,6 +3065,9 @@ func reconcileCAPIProviderDeployment(deployment *appsv1.Deployment, capiProvider
 	// Enforce pull policy.
 	for i := range deployment.Spec.Template.Spec.Containers {
 		deployment.Spec.Template.Spec.Containers[i].ImagePullPolicy = corev1.PullIfNotPresent
+		if deployment.Spec.Template.Spec.Containers[i].Name == "manager" {
```

Comment:
why is this not done in CAPIProviderDeploymentSpec for azure?

Reply:
Discussed in Slack. Moved to include this flag in `CAPIProviderDeploymentSpec` for Azure.